system/system_monitor/src/net_monitor/net_monitor.cpp

```diff
@@ -492,5 +502,136 @@ void NetMonitor::monitorTraffic(diagnostic_updater::DiagnosticStatusWrapper & st
+    stat.add(fmt::format("packet reassembles failed per unit time"), unit_reassembles_failed);
+
+    if (unit_reassembles_failed >= reassembles_failed_check_count_) {
+      whole_level = std::max(whole_level, static_cast<int>(DiagStatus::WARN));
+      error_str = "reassembles failed";
+    }
+
+    last_reassembles_failed_ = total_reassembles_failed;
+  } else {
+    reassembles_failed_queue_.push_back(0);
+    whole_level = std::max(whole_level, static_cast<int>(DiagStatus::ERROR));
+    error_str = "failed to read /proc/net/snmp";
+  }
+
+  if (!error_str.empty()) {
+    stat.summary(whole_level, error_str);
+  } else {
+    stat.summary(whole_level, "OK");
+  }
+
+  // Measure elapsed time since start time and report
+  SystemMonitorUtility::stopMeasurement(t_start, stat);
+}
+
+void NetMonitor::searchReassemblesFailedColumnIndex()
+{
+  std::ifstream ifs("/proc/net/snmp");
+  if (!ifs) {
+    RCLCPP_WARN(get_logger(), "Failed to open /proc/net/snmp.");
+    return;
+  }
+
+  // /proc/net/snmp
+  // Ip: Forwarding DefaultTTL InReceives ... ReasmTimeout ReasmReqds ReasmOKs ReasmFails ...
+  // Ip: 2          64         5636471397 ... 135          2303339    216166   270        ...
+  std::string line;
+
+  // Find column index of 'ReasmFails'
+  if (!std::getline(ifs, line)) {
+    RCLCPP_WARN(get_logger(), "Failed to get /proc/net/snmp first line.");
+    return;
+  }
+
+  std::vector<std::string> title_list;
+  boost::split(title_list, line, boost::is_space());
+
+  if (title_list.empty()) {
+    RCLCPP_WARN(get_logger(), "/proc/net/snmp first line is empty.");
+    return;
+  }
+  if (title_list[0] != "Ip:") {
+    RCLCPP_WARN(
+      get_logger(), "/proc/net/snmp line title column is invalid. : %s", title_list[0].c_str());
+    return;
+  }
+
+  int index = 0;
+  for (auto itr = title_list.begin(); itr != title_list.end(); ++itr, ++index) {
+    if (*itr == "ReasmFails") {
+      reassembles_failed_column_index_ = index;
+      break;
+    }
+  }
+}
+
+bool NetMonitor::getReassemblesFailed(uint64_t & reassembles_failed)
+{
+  if (reassembles_failed_column_index_ == 0) {
+    RCLCPP_WARN(
+      get_logger(), "reassembles failed column index is invalid. : %d",
+      reassembles_failed_column_index_);
+    return false;
+  }
+
+  std::ifstream ifs("/proc/net/snmp");
+  if (!ifs) {
+    RCLCPP_WARN(get_logger(), "Failed to open /proc/net/snmp.");
+    return false;
+  }
+
+  std::string line;
+
+  // Skip title row
+  if (!std::getline(ifs, line)) {
+    RCLCPP_WARN(get_logger(), "Failed to get /proc/net/snmp first line.");
+    return false;
+  }
+
+  // Find a value of 'ReasmFails'
+  if (!std::getline(ifs, line)) {
+    RCLCPP_WARN(get_logger(), "Failed to get /proc/net/snmp second line.");
+    return false;
+  }
+
+  std::vector<std::string> value_list;
+  boost::split(value_list, line, boost::is_space());
+
+  if (static_cast<std::size_t>(reassembles_failed_column_index_) >= value_list.size()) {
```

Comment:
You don't need static_cast.

Reply:
@ito-san 
I removed static_cast.
ffab6a9d97c1c6bcd7f805176c20578392e84b52